### DISCUSSION TITLE
zigbee2Tasmota legacy mode on some devices

### DISCUSSION THREAD

**Discussion by marcolino7** [arendst/Tasmota#17822] [Q&A]
Hi,
I am a newbie on zigbee and I need help.
I am trying to use Xiaomi WXKG01LM switch connected to Zigbee2Tasmota, it connect and I can receive MQTT message on my broker. Reporting of the button message is a nightmare to configure on OpenHab, because button send click instead action.
Reading a page on zigbee2mqtt site, I got some info on this button and I think there is what I need.
They say to disable "Legacy Mode" in order to get action event instead click event. here is the link of that page: 

https://www.zigbee2mqtt.io/devices/WXKG01LM.html

How I can achieve this with Tasmota?
Many thanks in advance for helping me.

**Comment by s-hadinger** [comment]
I'm afraid this Legacy mode is Z2M stuff, not a configuration of the device itself. 

Can you share logs of the messages received?

**Reply by marcolino7** [reply]
Hi,
Here is message i got from device when I press button.

Here is what I receive via MQTT on varius action:

**1 Click**
```
00:49:50.518 MQT: tele/ZigbeeBridge/087F/SENSOR = {"ZbReceived":{"0x087F":{"Device":"0x087F","Power":0,"Endpoint":1,"LinkQuality":225}}}
00:49:50.915 MQT: tele/ZigbeeBridge/087F/SENSOR = {"ZbReceived":{"0x087F":{"Device":"0x087F","Power":1,"Endpoint":1,"LinkQuality":211}}}
```
**Long Press**
```
00:51:42.356 MQT: tele/ZigbeeBridge/087F/SENSOR = {"ZbReceived":{"0x087F":{"Device":"0x087F","Power":0,"Endpoint":1,"LinkQuality":211}}}
00:51:43.758 MQT: tele/ZigbeeBridge/087F/SENSOR = {"ZbReceived":{"0x087F":{"Device":"0x087F","Power":1,"Endpoint":1,"LinkQuality":211}}}
```

**2 Click**
```
00:50:25.782 MQT: tele/ZigbeeBridge/087F/SENSOR = {"ZbReceived":{"0x087F":{"Device":"0x087F","Power":2,"Endpoint":1,"LinkQuality":222}}}
```
3 Click
```
00:50:55.446 MQT: tele/ZigbeeBridge/087F/SENSOR = {"ZbReceived":{"0x087F":{"Device":"0x087F","Power":3,"Endpoint":1,"LinkQuality":218}}}
```
4 Click
```
00:50:51.940 MQT: tele/ZigbeeBridge/087F/SENSOR = {"ZbReceived":{"0x087F":{"Device":"0x087F","Power":4,"Endpoint":1,"LinkQuality":222}}}
```

As you can see, only difference between One Click and Long press, is the time between two messages on MQTT.

Reading Z2M articles, seem that it is possible to set the sensor Options. In case of the xiaomi Switch I need to set  Legacy to change event from click to actions, but Options are used also to setup sensor behaviour, in example this occupancy sensor:
https://www.zigbee2mqtt.io/devices/RTCGQ14LM.html
with the Options I can set occupancy_timeout or calibrate the illuminance.

I would like to understand if it is possible to achieve this in Tasmota

Marco

**Comment by Jason2866** [comment]
Installed this Zigbee button in my OpenHab this week. Did a rule in OpenHab to detect the change and the rule fires different actions on the value of `Power`.  Don't see where it is a Nightmare. Whole setup was done in a few minutes.
Don't use long press at all. It is always a pain to use.

**Reply by marcolino7** [reply]
Hi,
I need long press, I am migrating from Xiaomi Controller, and I need Long  Press.

**Comment by marcolino7** [comment]
I Installed Z2M, and this is option I need:

![image](https://user-images.githubusercontent.com/11460036/215341147-1f82a7b0-c0dc-4b55-95a2-12e910337bd8.png)
